analyzers/src/SonarAnalyzer.Common/Helpers/SonarLintXmlReader.cs

```diff
@@ -28,16 +28,20 @@ namespace SonarAnalyzer.Helpers;
 
 public class SonarLintXmlReader
 {
-    public static readonly SonarLintXmlReader Empty = new(null, LanguageNames.CSharp);
+    public static readonly SonarLintXmlReader Empty = new(null);
 
     private readonly SonarLintXml sonarLintXml;
     private readonly string propertyLanguage;
 
     private bool? ignoreHeaderComments;
-    public bool IgnoreHeaderComments => ignoreHeaderComments ??= ReadBoolean(ReadSettingsProperty($"sonar.{propertyLanguage}.ignoreHeaderComments"));
+    private bool IgnoreHeaderCommentsCS => ignoreHeaderComments ??= ReadBoolean(ReadSettingsProperty($"sonar.cs.ignoreHeaderComments"));
+    private bool IgnoreHeaderCommentsVB => ignoreHeaderComments ??= ReadBoolean(ReadSettingsProperty($"sonar.vbnet.ignoreHeaderComments"));
+    public bool IgnoreHeaderComments => IgnoreHeaderCommentsCS || IgnoreHeaderCommentsVB;
```

Comment:
This doesn't make sense, there needs to be public `IgnoreHeaderCommentsCS` and `IgnoreHeaderCommentsVB` where the caller does the switch expression on the language and chooses.

Or even better - there needs to be a public method `IgnoreHeaderComments(string Langauge)` that will do the decision inside